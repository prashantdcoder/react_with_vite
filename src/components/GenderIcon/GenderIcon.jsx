/* eslint-disable no-unused-vars */
import PropTypes from "prop-types";
import React from "react";

const GenderIcon = ({ gender }) => {
  return gender === "male" ? (
    <span style={{ fontSize: "20px" }}>&#128104;</span>
  ) : (
    <span style={{ fontSize: "20px" }}>&#128105;</span>
  );
};

GenderIcon.propTypes = {
  gender: PropTypes.string.isRequired,
};

export default GenderIcon;
