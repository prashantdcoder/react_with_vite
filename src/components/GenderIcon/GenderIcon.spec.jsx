import { render } from "@testing-library/react";
import React from "react";
import GenderIcon from "./GenderIcon";

describe("GenderIcon component", () => {

  it("renders the female gender icon correctly", () => {
    const { asFragment } = render(<GenderIcon gender="female" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the male gender icon correctly", () => {
    const { asFragment } = render(<GenderIcon gender="male" />);
    expect(asFragment()).toMatchSnapshot();
  });

});
