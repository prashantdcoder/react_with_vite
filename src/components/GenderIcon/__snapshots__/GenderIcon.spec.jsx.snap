// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenderIcon component renders the female gender icon correctly 1`] = `
<DocumentFragment>
  <span
    style="font-size: 20px;"
  >
    ðŸ‘©
  </span>
</DocumentFragment>
`;

exports[`GenderIcon component renders the male gender icon correctly 1`] = `
<DocumentFragment>
  <span
    style="font-size: 20px;"
  >
    ðŸ‘¨
  </span>
</DocumentFragment>
`;
